service/resource/project/resource.go

```diff
@@ -101,7 +102,7 @@ func (r *Resource) GetDesiredState(obj interface{}) (interface{}, error) {
 	var desiredProjects []Project
 
 	for _, p := range customObject.Spec.Projects {
-		desiredProjects = append(desiredProjects, Project{Name: p.Name, Ref: p.Ref})
+		desiredProjects = append(desiredProjects, Project{ID: p.ID, Name: p.Name, Ref: p.Ref})
```

Comment:
Note that the list of current projects does not contain any IDs because helm does not know about them.